primitives/src/merkle_tree/field_based_mht/optimized/mod.rs

```diff
@@ -41,23 +41,24 @@ impl<T: BatchFieldBasedMerkleTreeParameters> FieldBasedOptimizedMHT<T> {
     /// `primitives/src/benches/poseidon_mht.rs` to properly tune the `processing_step`
     /// parameter according to your use case.
     pub fn init(height: usize, processing_step: usize) -> Result<Self, Error> {
-        // TODO: check error message
         if !check_precomputed_parameters::<T>(height) {
-            return Err(Box::new(MerkleTreeError::Other("precomputed parameters check failed".to_owned())));
+            Err(Box::new(MerkleTreeError::Other(format!(
+                "Unsupported height. Max supported height is: {}",
+                T::ZERO_NODE_CST.unwrap().nodes.len()
+            ).to_owned())))?
         }
 
         let rate = <<T::H as FieldBasedHash>::Parameters as FieldBasedHashParameters>::R;
         // Rate may also be smaller than the arity actually, but this assertion
         // is reasonable and simplify the design.
-        // TODO: check error message
-        if rate != T::MERKLE_ARITY {
-            return Err(Box::new(MerkleTreeError::Other("rate check failed".to_owned())));
-        }
+        assert_eq!(rate, T::MERKLE_ARITY);
 
         let last_level_size = T::MERKLE_ARITY.pow(height as u32);
-        // TODO: check error message
         if processing_step <= 0 || processing_step > last_level_size {
```

Comment:
We can remove the check <= 0 since 'processing_step' is an usize, and replace it with  processing_step == 0